### ISSUE TITLE
Config generator doesn't match nullability semantics of built-in Bind

### ISSUE THREAD

**Issue by stephentoub** [dotnet/runtime#90987]
The built-in Bind method annotates the object as nullable:
```C#
public static void Bind(this IConfiguration configuration, object? instance)
```
and then explicitly nops if it's null:
```C#
if (instance != null)
{
    ...
}
```
In contrast, the code generated by the generator annotates it as non-nullable and throws an exception if it's null, changing the semantics when using the source generator.
```C#
if (obj is null)
{
    throw new ArgumentNullException(nameof(obj));
}
```
(It also names the parameter `obj` instead of `instance`.)

**Comment by layomia** [comment]
Consistency is desired cc @tarekgh.

**Comment by layomia** [comment]
Fixed for 8.0 in https://github.com/dotnet/runtime/pull/91359.
